proto/pulumi/resource_status.proto

```diff
@@ -0,0 +1,64 @@
+// Copyright 2025, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+
+import "google/protobuf/struct.proto";
+import "pulumi/provider.proto";
+
+package pulumirpc;
+
+option go_package = "github.com/pulumi/pulumi/sdk/v3/proto/go;pulumirpc";
+
+// ResourceStatus is an interface that can be called from a resource provider to update status about a resource.
+service ResourceStatus {
+    // `PublishViewSteps` is used to publish a series of steps for a view resource.
+    // Views can be materialized via create and update steps, and more complex
+    // changes, such as replacements, can be modeled as a series of steps.
+    // The engine does not actually apply these steps, but rather flows them through
+    // the engine such that the view resources are written to state and the view
+    // resources are displayed in the UI.
+    rpc PublishViewSteps(PublishViewStepsRequest) returns (PublishViewStepsResponse) {}
+}
+
+// `PublishViewStepsRequest` is the type of requests sent as part of a
+// [](pulumirpc.ResourceStatus.PublishViewSteps) call.
+message PublishViewStepsRequest {
+    string token = 1; // the service context token
+    string type = 2; // the type of the view resource
+    string name = 3; // the name, for URN purposes, of the view resource
```

Comment:
Not allowing sub-parenting? Everything is just directly under the parent custom resource

Reply:
I was going to initially punt on supporting this, but maybe we should just go ahead and do it.

I was trying to avoid exposing view URNs, instead only exposing type/name (and computing the URN internally).

If we wanted to support `parent`, I think we'd probably want to expose that as accepting a URN, which means we'd have to return the view's URN, probably from `PublishViewStepsResponse`.

We'd probably want to validate that the URN is either the parent custom resource or one of the sub views.